app/models/user.php

```diff
@@ -0,0 +1,67 @@
+<?php
+class User extends AppModel
+{
+    const ERR_DUPLICATE_ENTRY = 1062;
+    public $user_validated = true;
+    public $validation = array(
+        'username' => array(
+            'length' => array(
+                'validate_between', 1, 16,
+            ),
+        ),
+        'password' => array(
+            'length' => array(
+                'validate_between', 1, 16,
+            ),
+        ),
+    );
+
+    public function login(User $user)
+    {
+        if (!$this->isUserExisting() || !$this->isPasswordMatch()) {
+            $this->user_validated = false;
+        }
```

Comment:
I don't think this check is needed since fetching the user data below is enough for login check.